### PR TITLE
feat: CvCheckbox & CvLoading vue3

### PR DESCRIPTION
**Pull request by davidnixon**
Contributes to Vue 3

## What did you do?
Port checkbox and loading from Vue 2. Replace check mixin with cvCheck composable

## How have you tested it?
Tests are not yet complete but they new use the new [vue-testing-library](https://github.com/testing-library/vue-testing-library)

## Were docs updated if needed?

- [ ] N/A
- [ ] No
- [ ] Yes

### CODE REVIEW

File: src/components/CvCheckbox/__tests__/CvCheckbox.spec.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import CvCheckbox from '../CvCheckbox';
+
+describe('CvCheckbox', () => {
+  it('CvCheckbox - default', async () => {
+    // The render method returns a collection of utilities to query your component.
+    const { emitted, getByTestId } = render(CvCheckbox, {
+      props: { value: 'checkbox-1' },
+    });
+
+    const checkbox = getByTestId('checkbox');
```
Comment (lee-chase) [inline src/components/CvCheckbox/__tests__/CvCheckbox.spec.js:11]: I would ordinarily only use getByTestId as a last resort. In this case, and the next,  getByRole would be more appropriate

File: src/components/CvCheckbox/__tests__/CvCheckbox.spec.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    await fireEvent.click(checkbox);
```
Comment (lee-chase) [inline src/components/CvCheckbox/__tests__/CvCheckbox.spec.js:13]: For many events UserEvent preferred to fireEvent https://testing-library.com/docs/user-event/

```
import userEvent from '@testing-library/user-event';
...
await userEvent.click(checkbox);
...
```

File: src/components/CvCheckbox/__tests__/CvCheckbox.spec.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      label: 'CHECKBOX XYZ123',
+      id: 'ABC-checkbox-123',
+      checked: true,
+    };
+    // The render method returns a collection of utilities to query your component.
+    const { emitted, getByText, getByTestId } = render(CvCheckbox, {
+      props,
+    });
+
+    const label = getByText(props.label);
```
Comment (lee-chase) [inline src/components/CvCheckbox/__tests__/CvCheckbox.spec.js:30]: No need to save `label` as getByTtext includes an expect.

File: src/components/CvCheckbox/__tests__/CvCheckbox.spec.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+    const checkbox = getByTestId('checkbox');
```
Comment (lee-chase) [inline src/components/CvCheckbox/__tests__/CvCheckbox.spec.js:32]: Use getByRole

File: src/components/CvCheckbox/__tests__/CvCheckbox.spec.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+  it('CvCheckbox - props', async () => {
+    const props = {
+      value: 'checkbox-1',
+      label: 'CHECKBOX XYZ123',
+      id: 'ABC-checkbox-123',
+      checked: true,
+    };
+    // The render method returns a collection of utilities to query your component.
+    const { emitted, getByText, getByTestId } = render(CvCheckbox, {
```
Comment (lee-chase) [inline src/components/CvCheckbox/__tests__/CvCheckbox.spec.js:26]: I needed to use `= await render(` in order to pass the `checkbox.value` test later.

**Review by lee-chase — COMMENTED** [review]
In addition to other comments, I think a component should always have a test for `$attrs`, `$listeners`,  `class`. Some ma

File: src/components/CvCheckbox/__tests__/CvCheckbox.spec.js (hunk 2 above)
Comment (davidnixon) [reply src/components/CvCheckbox/__tests__/CvCheckbox.spec.js]: done. add the `@testing-library/user-event` package and the requested peer package `@testing-library/dom`

File: src/components/CvCheckbox/__tests__/CvCheckbox.spec.js (hunk 3 above)
Comment (davidnixon) [reply src/components/CvCheckbox/__tests__/CvCheckbox.spec.js]: fixed

File: src/components/CvCheckbox/__tests__/CvCheckbox.spec.js (hunk 1 above)
Comment (davidnixon) [reply src/components/CvCheckbox/__tests__/CvCheckbox.spec.js]: `$listeners` combined into `attrs` in Vue3 of course. I added test for `@click` listener on the checkbox and set an additional attribute to test those.

**Comment by lee-chase** [comment]
Looks good to me @davidnixon 

Feel free to go ahead and merge.

**Pull request merged**